extension/src/openvic-extension/utility/StringLiteral.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <string_view>
+#include <type_traits>
+
+namespace OpenVic {
+	template<std::size_t N>
+	struct StringLiteral {
+		constexpr StringLiteral(const char (&str)[N]) {
+			std::copy_n(str, N, value);
+		}
+
+		constexpr StringLiteral(std::string_view string) {
+			std::copy_n(string.begin(), N, value);
```

Comment:
This will happily copy from data after the end of the string view, which may fault, depending where the string view is pointing

Reply:
Should be fixed running as consteval and having an assert that ensures `string.size() == N`